### PR TITLE
Martin/jib 179 blur background behind the thread

### PR DESCRIPTION
**Pull request by MartinJibber**
Adding a blur background behind the header in the thread VC.

Automatically scrolling to the latest messages in the thread VC
Simulator Screen Shot - iPhone 12 - 2021-11-01 at 15 17 02

### CODE REVIEW

File: Shared/Models/AnimationCycle.swift [hunk 1]
```diff
@@ -13,4 +13,5 @@ struct AnimationCycle {
     var outToPosition: AnimationPosition
     var shouldConcatenate: Bool
     var scrollToIndexPath: IndexPath?
+    var scrollPosition: UICollectionView.ScrollPosition = .centeredHorizontally
```
Comment (MartinJibber) [inline Shared/Models/AnimationCycle.swift:16]: Auto scroll behavior needs to specify the scroll position.

File: iOS/Collection View Management /CollectionViewDataSource.swift [hunk 2]
```diff
@@ -347,10 +347,12 @@ extension CollectionViewDataSource {
...
 
-        // If specified, scroll to the last item in the collection view.
-        if let scrollToIndexPath = animationCycle.scrollToIndexPath {
+        // If specified, scroll to a particular item in the collection view.
+        if let scrollToIndexPath = animationCycle.scrollToIndexPath,
+           self.itemIdentifier(for: scrollToIndexPath).exists {
+
             collectionView.scrollToItem(at: scrollToIndexPath,
-                                        at: [.centeredHorizontally],
+                                        at: [animationCycle.scrollPosition],
```
Comment (MartinJibber) [inline iOS/Collection View Management /CollectionViewDataSource.swift:355]: Now the animation cycle scroll is customized.

File: iOS/Flows/Conversation/Threads/ConversationThreadViewController.swift [hunk 3]
```diff
@@ -18,6 +18,8 @@ class ConversationThreadViewController: DiffableCollectionViewController<Convers
 
     private let blurView = UIVisualEffectView(effect: UIBlurEffect(style: .dark))
     private let parentMessageView = ThreadMessageCell()
+    /// A view positioned behind the parent message to separate it from the rest of the messages.
+    private let parentMessageBlurView = UIVisualEffectView(effect: UIBlurEffect(style: .regular))
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Threads/ConversationThreadViewController.swift:22]: Just a basic blur view behind the header.

File: iOS/Flows/Conversation/Threads/ConversationThreadViewController.swift [hunk 4]
```diff
@@ -85,8 +88,13 @@ class ConversationThreadViewController: DiffableCollectionViewController<Convers
...
         self.blurView.expandToSuperviewSize()
 
+        self.parentMessageBlurView.expandToSuperviewWidth()
+        self.parentMessageBlurView.height = 120
+        self.parentMessageBlurView.roundCorners()
+
         self.parentMessageView.width = self.view.width * 0.8
-        self.parentMessageView.height = 120
+        self.parentMessageView.height = 100
+        self.parentMessageView.pin(.top, padding: 10)
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Threads/ConversationThreadViewController.swift]: Use Theme.contentOffset everywhere.

File: iOS/Flows/Conversation/Threads/ConversationThreadViewController.swift [hunk 5]
```diff
@@ -85,8 +88,13 @@ class ConversationThreadViewController: DiffableCollectionViewController<Convers
 
         self.blurView.expandToSuperviewSize()
 
+        self.parentMessageBlurView.expandToSuperviewWidth()
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Threads/ConversationThreadViewController.swift]: Make this width - Theme.contentOffset.doubled

File: iOS/Flows/Conversation/Threads/ConversationThreadViewController.swift [hunk 6]
```diff
@@ -137,7 +145,11 @@ class ConversationThreadViewController: DiffableCollectionViewController<Convers
     }
 
     override func getAnimationCycle() -> AnimationCycle? {
-        return nil
+        var cycle = super.getAnimationCycle()
+        cycle?.shouldConcatenate = false
+        cycle?.scrollToIndexPath = IndexPath(item: 0, section: 0)
+        cycle?.scrollPosition = .bottom
+        return cycle
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Threads/ConversationThreadViewController.swift:155]: Should concatenate is false. This animates all the cells at the same time so the vertical line doesn't look weird.

Also scrolling to the first message which is the most recent one.

**Review by Benjamindodgson — CHANGES_REQUESTED** [review]
Just use Theme.contentOffset

**Pull request merged**